### PR TITLE
Add placeholder support for e2e and some reclaim cases

### PR DESCRIPTION
**Pull request by alcorj-mizar**

### CODE REVIEW

File: test/e2e/reclaim.go [hunk 1]
```diff
@@ -6,10 +6,180 @@ import (
...
 )
 
 var _ = Describe("Queue E2E Test", func() {
+	It("Reclaim: New queue with job created no reclaim when resource is enough", func() {
+		q1 := "default"
+		q2 := "reclaim-q2"
+		ctx := initTestContext(options{
+			queues:             []string{q2},
+			nodesNumLimit:      4,
+			nodesResourceLimit: v1.ResourceList{"cpu": resource.MustParse("1000m"), "memory": resource.MustParse("1024Mi")},
```
Comment (hzxuzhonghu) [inline test/e2e/reclaim.go:23]: nit: can we use oneCPU or something?

Comment (alcorj-mizar) [reply test/e2e/reclaim.go]: small nits fixed

File: test/e2e/util.go [hunk 2]
```diff
@@ -889,6 +911,142 @@ func waitReplicaSetReady(ctx *testContext, name string) error {
...
+	if num <= len(nodes.Items)-taintsNodes {
+		return true
+	}
+	return false
+}
+
+func setPlaceHolderForSchedulerTesting(ctx *testContext, req v1.ResourceList, reqNum int) (bool, error) {
+
+	if !satisifyMinNodesRequirements(ctx, reqNum) {
+		return false, lagencyerror.New("Failed to setup environment, you need to have at least " + strconv.Itoa(len(req)) + " worker node.")
```
Comment (hzxuzhonghu) [inline test/e2e/util.go:933]: nit: fmt.Errorf

File: test/e2e/util.go [hunk 3]
```diff
@@ -889,6 +911,142 @@ func waitReplicaSetReady(ctx *testContext, name string) error {
...
+			},
+			NodeName: nodeName,
+		},
+	}
+	_, err := ctx.kubeclient.CoreV1().Pods("default").Create(context.TODO(), pod, metav1.CreateOptions{})
+	return err
+}
+
+func deletePlaceHolder(ctx *testContext) {
+	podList, err := ctx.kubeclient.CoreV1().Pods("default").List(context.TODO(), metav1.ListOptions{})
```
Comment (hzxuzhonghu) [inline test/e2e/util.go:1040]: nit: s/"default"/ctx.Namespace/

same applies to create place holder

File: test/e2e/util.go [hunk 4]
```diff
@@ -889,6 +911,142 @@ func waitReplicaSetReady(ctx *testContext, name string) error {
...
+	Expect(err).NotTo(HaveOccurred())
+	for _, pod := range podList.Items {
+		if pod.Labels["role"] == "placeholder" {
```
Comment (hzxuzhonghu) [inline test/e2e/util.go:1043]: This is not efficient loop all the pods, you can use field LabelSelector instead

File: test/e2e/reclaim.go [hunk 5]
```diff
@@ -7,9 +7,178 @@ import (
...
 )
 
 var _ = Describe("Queue E2E Test", func() {
+	It("Reclaim: New queue with job created no reclaim when resource is enough", func() {
+		q1 := "default"
+		q2 := "reclaim-q2"
+		ctx := initTestContext(options{
+			queues:             []string{q2},
+			nodesNumLimit:      4,
+			nodesResourceLimit: CPU1Mem1,
```
Comment (hzxuzhonghu) [inline test/e2e/reclaim.go:21]: nit: How could you know how much resource and how many nodes to occupy?

Comment (alcorj-mizar) [reply test/e2e/reclaim.go]: According to test cases, and there is a node num check in init processes, the check will tell user to skip some cases that they can not afford.

File: test/e2e/util.go (hunk 4 above)
Comment (alcorj-mizar) [reply test/e2e/util.go:1041]: OK

**Pull request merged**